x-pack/plugins/apm/public/components/shared/key_value_filter_list/index.tsx

```diff
@@ -78,38 +79,40 @@ export function KeyValueFilterList({
       buttonClassName="buttonContentContainer"
     >
       <StyledEuiDescriptionList type="column">
-        {nonEmptyKeyValueList.map(({ key, value }) => {
+        {nonEmptyKeyValueList.map(({ key, value, showFilterByOption }) => {
```

Comment:
nit: previous code does this:

```ts
  const nonEmptyKeyValueList = removeEmptyValues(keyValueList);
  if (!nonEmptyKeyValueList.length) {
    return null;
  }
```


What we want is actually:
```ts
  const hasNonEmptyValues = keyValueList.some(({ value }) => value !== undefined);
  if (!hasNonEmptyValues) {
    return null;
  }
```

That is faster and more idiomatic.

Reply:
Oh, nvm. We actually do want to filter out the non-empty elements. Disregard this comment.